From 8ecbb57a730c20505e6117e8c5f012ed3348e1ec Mon Sep 17 00:00:00 2001
From: gaulthier gain <gaulthier.gain@uliege.be>
Date: Fri, 9 Oct 2020 15:57:11 +0000
Subject: [PATCH] Update stdoutwrite by removing ioctl

Signed-off-by: gaulthier gain <gaulthier.gain@uliege.be>
---
 src/stdio/__stdout_write.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/stdio/__stdout_write.c b/src/stdio/__stdout_write.c
index dd1ec60..70dc450 100644
--- a/src/stdio/__stdout_write.c
+++ b/src/stdio/__stdout_write.c
@@ -5,7 +5,7 @@ size_t __stdout_write(FILE *f, const unsigned char *buf, size_t len)
 {
 	struct winsize wsz;
 	f->write = __stdio_write;
-	if (!(f->flags & F_SVB) && __syscall(SYS_ioctl, f->fd, TIOCGWINSZ, &wsz))
+	if (!(f->flags & F_SVB))
 		f->lbf = -1;
 	return __stdio_write(f, buf, len);
 }
-- 
2.17.1

