--- a/src/thread/x86_64/__set_thread_area.s	2018-02-22 19:39:19.000000000 +0100
+++ b/src/thread/x86_64/__set_thread_area.s	2020-11-13 12:26:36.600544392 +0100
@@ -3,8 +3,6 @@
 .global __set_thread_area
 .type __set_thread_area,@function
 __set_thread_area:
-	mov %rdi,%rsi           /* shift for syscall */
-	movl $0x1002,%edi       /* SET_FS register */
-	movl $158,%eax          /* set fs segment to */
-	syscall                 /* arch_prctl(SET_FS, arg)*/
+	wrfsbase %rdi
+       movq $0,%rax
 	ret
